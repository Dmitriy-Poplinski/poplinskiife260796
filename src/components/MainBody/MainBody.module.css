.main_body {
    flex-grow: 3;
    display: flex;
    align-items: center;
    justify-content: center;
};
