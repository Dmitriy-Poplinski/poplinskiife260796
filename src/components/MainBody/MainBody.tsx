import React from 'react';

import { MyButton } from '../Button/MyButton';

import classes from './MainBody.module.css';

interface MainBodyProps {
    image: string
    shuffleImageArray: () => void
}

export function MainBody({ image, shuffleImageArray }: MainBodyProps) {
    let color;

    try {
        color = `url(${require(`../../assets/images/${image}.jpg`)})`; 
    } catch (e) {
        color = 'violet'
    };

    return (
        <div 
            className={classes.main_body}
            style={{
                background: color
            }}
            data-testid="main_body"
        >
            <MyButton shuffleImageArray={shuffleImageArray}/>
        </div>
    );
}
