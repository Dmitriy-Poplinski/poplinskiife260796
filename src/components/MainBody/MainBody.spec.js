import { render, screen } from '@testing-library/react';

import { MainBody } from './MainBody';

describe('MainBody:', () => {

    let data; 

    test('MainBody gets incorrect string', () => {
        data = 'testing_image';

        render(<MainBody image={data} />);

        expect(screen.getByTestId('main_body')).toHaveStyle('background: violet');
        expect(screen.getByTestId('main_body')).toMatchSnapshot();
    });

    test('MainBody gets correct string', () => {
        data = 'main_body_cover';

        render(<MainBody image={data} />);

        expect(screen.getByTestId('main_body')).toHaveStyle(`background: url("main_body_cover.jpg")`);
        expect(screen.getByTestId('main_body')).toMatchSnapshot();
    });
});
