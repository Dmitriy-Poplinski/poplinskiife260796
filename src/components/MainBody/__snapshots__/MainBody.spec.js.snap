// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainBody: MainBody gets correct string 1`] = `
<div
  class="main_body"
  data-testid="main_body"
  style="background: url(main_body_cover.jpg);"
>
  <button
    class="button"
    data-testid="button"
  >
    Click me
  </button>
</div>
`;

exports[`MainBody: MainBody gets incorrect string 1`] = `
<div
  class="main_body"
  data-testid="main_body"
  style="background: violet;"
>
  <button
    class="button"
    data-testid="button"
  >
    Click me
  </button>
</div>
`;
